reconcile/utils/terrascript_client.py

```diff
@@ -3155,6 +3175,154 @@ def is_elasticsearch_domain_name_valid(name):
+                    region = ns['cluster']['spec']['region']
+                    account = res['account']
+                    account_id = self.accounts[account]['uid']
+                    lg_identifier = \
+                        TerrascriptClient.elasticsearch_log_group_identifier(
+                            domain_identifier=res['identifier'],
+                            log_type=ElasticSearchLogGroupType(log_type),
+                        )
+                    log_group_infos.append(
+                        ElasticSearchLogGroupInfo(
+                            account=account,
+                            account_id=account_id,
+                            region=region,
+                            log_group_identifier=lg_identifier,
+                        )
+                    )
+        return log_group_infos
+
+    def _get_tf_resource_elasticsearch_resource_policy(
+        self, account: str
+    ) -> aws_cloudwatch_log_resource_policy:
+        '''
+        https://docs.aws.amazon.com/opensearch-service/latest/developerguide/createdomain-configure-slow-logs.html
+        CloudWatch Logs supports 10 resource policies per Region.
+        If you plan to enable logs for several OpenSearch Service domains,
+        you should create and reuse a broader policy that includes multiple
+        log groups to avoid reaching this limit.
+        I.e., ideally we aggregate ALL log group identifiers for each
+        account first.
+        '''
+        log_group_infos = \
+            self._elasticsearch_aggregate_log_groups_per_account()
+
+        log_groups_policy = {
+            'Version': '2012-10-17',
+            'Statement': [{
+                'Effect': 'Allow',
+                'Principal': {
+                    'Service': 'es.amazonaws.com',
+                },
+                'Action': [
+                    'logs:PutLogEvents',
+                    'logs:CreateLogStream',
+                ],
+                'Resource': [
+                    (
+                        f'arn:aws:logs:{info.region}:{info.account_id}'
+                        f':log-group:{info.log_group_identifier}:*'
+                    )
+                    for info in log_group_infos if info.account == account
+                ],
+            }]
+        }
+        log_groups_policy_values = {
+            'policy_name': 'es-log-publishing-permissions',
+            'policy_document': json.dumps(log_groups_policy, sort_keys=True),
+        }
+        resource_policy = aws_cloudwatch_log_resource_policy(
+            'es_log_publishing_resource_policy',
+            **log_groups_policy_values,
+        )
+        return resource_policy
+
+    def _get_tf_resource_elasticsearch_log_groups(
+        self, identifier: str, account: str,
+        resource: Mapping[str, Any], values: Mapping[str, Any],
+        output_prefix: str
+    ) -> tuple[list[dict[str, Any]], list[dict[str, str]]]:
+        ES_LOG_GROUP_RETENTION_DAYS = 180
```

Comment:
Can someone confirm that 180 days retention period makes sense here?

Reply:
can you make 180 the default and also allow to define this value in app-interface?

Reply:
Reduced it to 90 days now. 

I am wondering if for `AUDIT_LOGS` a longer period might be necessary. Anyways that can be tackled in the follow-up ticket dedicated to `AUDIT_LOGS`.